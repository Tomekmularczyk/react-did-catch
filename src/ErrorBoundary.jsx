import React from 'react';
import Types from 'prop-types';
import ErrorMessage from './ErrorMessage';
import browserInfo from './browserInfo';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      info: null,
      hasError: false,
    }
  }

  componentDidCatch(error, info) {
    this.setState({ info, error, hasError: true });
  }

  render() {
    const { error, info, hasError } = this.state;
    const { render } = this.props;
    if (hasError) {
      if (typeof (render) === 'function') {
        return render(error, info, browserInfo);
      } else {
        return (
          <ErrorMessage
            error={error}
            info={info}
          />
        );
      }
    }

    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  render: Types.func,
};

export default ErrorBoundary;