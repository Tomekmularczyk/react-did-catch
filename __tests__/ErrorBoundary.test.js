import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ErrorBoundary from '../src/ErrorBoundary'

describe('<ErrorBoundary />', () => {
  let component
  beforeEach(() => {
    component = shallow(
      <ErrorBoundary isExpanded={false}>children</ErrorBoundary>
    )
  })

  it('matches snapshots', () => {
    expect(toJson(component)).toMatchSnapshot()

    component.setState({ hasError: true, error: new Error(''), info: { componentStack: '' } })
    expect(toJson(component)).toMatchSnapshot()
  })
})
