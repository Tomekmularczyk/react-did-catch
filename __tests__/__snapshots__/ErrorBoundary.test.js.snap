// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> matches snapshots 1`] = `"children"`;

exports[`<ErrorBoundary /> matches snapshots 2`] = `
<ErrorMessage
  error={[Error]}
  info={
    Object {
      "componentStack": "",
    }
  }
/>
`;
